ddclient.in

```diff
@@ -2370,6 +2369,39 @@ sub extract_ipv6 {
     return $ip;
 }
 
+######################################################################
+## Regex that matches an IPv6 global address. Accepts embedded leading zeros.
+## A Global Unicast Address (GUA) is any address in a range not otherwise
+## reserved.  So rule out any ULA (FC00::/7), multicast (FF00::/8) or
+## link local (FE80::/10) addresses. We also require that it start
+## with a non-zero /16 prefix though strictly a GUA could.
+######################################################################
+my $regex_ipv6_global = qr/
+    (?!FF[A-F0-9]{2}:)      # Not a multicast
+    (?!F[CD][A-F0-9]{2}:)   # And not a ULA
+    (?!FE[89AB][A-F0-9]:)   # And not a link local
+    (?!0{0,4}:)             # And does not start with zeros
```

Comment:
> Perl is misbehaving with comments embedded within regex... cannot have anything that looks like a regex directive...

It's misbehaving because you're using `qr/.../`. Perl sees the slash in the comment as an end-of-regex marker. Try something like `qr{...}`, `qr@...@`, or `qr%...%` and it should work.

> simply testing for :: or ::1 is not good enough, you need 0000:/16

Why?

Reply:
> > Perl is misbehaving with comments embedded within regex... cannot have anything that looks like a regex directive...
> 
> It's misbehaving because you're using `qr/.../`. Perl sees the slash in the comment as an end-of-regex marker. Try something like `qr{...}`, `qr@...@`, or `qr%...%` and it should work.

Thank you, that fixed it.